kernel/src/paging.c

```diff
@@ -39,16 +80,44 @@ const PageFlags pagingmasks[5] = {
     (1L << 48) - (1L << 39)  // P4: bits (48 - 39]
 };
 
+// returns if a pagetable entry can be expressed as a large page, rather than a table
+bool check_table_redundancy(PageEntry page_entry, uint64_t page_depth) {
```

Comment:
I still don't think we should do this automatically. This is a very rare event, but can be a bit expensive to do every time. It isn't worth it in my opinion. It could also cause problems when multiple cores are using the same page table.  We could keep it as something to do manually.

Reply:
This would be worthwhile for repeated usage of the same pagetable, but for the case of creating and discarding pagetables for each process it would indeed be unreasonable to include.